### PR TITLE
Support Compose model feature FR

### PR DESCRIPTION
**Pull request by samvaity**

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingAsyncClient.java [hunk 1]
```diff
@@ -470,6 +472,17 @@ String getEndpoint() {
         }
     }
 
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public PollerFlux<FormRecognizerOperationResult, CustomFormModel> beginCreateCompositeModel(List<String> modelIds) {
```
Comment (samvaity) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingAsyncClient.java:476]: beginCreateComposedModel

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/models/CustomModelProperties.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.azure.ai.formrecognizer.training.models;
+
+public class CustomModelProperties {
```
Comment (samvaity) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/models/CustomModelProperties.java]: final

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/models/CreateCompositeModelOptions.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package com.azure.ai.formrecognizer.training.models;
+
+import java.time.Duration;
+
+public class CreateCompositeModelOptions {
```
Comment (samvaity) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/models/CreateCompositeModelOptions.java]: fluent, final

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingAsyncClient.java (hunk 1 above)
Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingAsyncClient.java:476]: Could this be Set?

Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingAsyncClient.java:476]: Iterable?

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/models/CustomModelProperties.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+    private boolean compositeModel;
+
+    public boolean isCompositeModel() {
+        return this.compositeModel;
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/models/CustomModelProperties.java:13]: nit. `this` can be removed

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/models/TrainingDocumentInfo.java [hunk 5]
```diff
@@ -14,6 +14,7 @@
  */
 @Immutable
 public final class TrainingDocumentInfo {
+    private String modelId;
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/models/TrainingDocumentInfo.java:17]: a new constructor needed to set `modelId`

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/models/CustomFormModel.java [hunk 6]
```diff
@@ -16,6 +16,9 @@
 @Immutable
 public final class CustomFormModel {
 
+    private String displayName;
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/models/CustomFormModel.java:19]: Another new constructor is needed to assign value to `displayName` and `modelProperties`

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/models/TrainingDocumentInfo.java (hunk 5 above)
Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/models/TrainingDocumentInfo.java:17]: Same as this https://github.com/mssfang/azure-sdk-for-java/pull/41#discussion_r490439055

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/models/CustomFormSubmodel.java [hunk 7]
```diff
@@ -14,6 +14,8 @@
 @Immutable
 public final class CustomFormSubmodel {
 
+    private String modelId;
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/models/CustomFormSubmodel.java:17]: Another new constructor is needed to assign value to `modelId`

File: sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/CreateCompositeModel.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        FormTrainingClient client = new FormTrainingClientBuilder()
+            .credential(new AzureKeyCredential("{key}"))
+            .endpoint("https://{endpoint}.cognitiveservices.azure.com/")
+            .buildClient();
+
+        List<String> modelIdList = Arrays.asList("modelId1", "modelId2");
+        final CustomFormModel customFormModel =
+            client.beginCreateCompositeModel(modelIdList).getFinalResult();
+
+        System.out.printf("Display name for model: %s", customFormModel.getDisplayName());
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/CreateCompositeModel.java:23]: do we want to show the compositeModel's model id as well?

**Review by mssfang — APPROVED** [review]

**Pull request merged**